flocker/control/httpapi.py

```diff
@@ -185,6 +185,63 @@ def saved(ignored):
         saving.addCallback(saved)
         return saving
 
+    # Will we also have a /state endpoint? Or just route directly to the datasets path?
+    @app.route("/state/datasets", methods=['GET'])
+    @user_documentation("""
+        Get current cluster datasets
+        """, examples=[u"get state datasets"])
+    @structured(
+        inputSchema={},
+        outputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets_array'},
+        schema_store=SCHEMAS
+    )
+    def datasets(self):
+        """
+        Return the current primary datasets in the cluster.
+        """
+        deployment = self.cluster_state_service.as_deployment()
+        return list(datasets_from_deployment(deployment))
+
+
+def datasets_from_deployment(deployment):
+    """
+    Extract the primary datasets from the supplied deployment instance.
+
+    XXX: What would be the point in returning secondary dataset dictionaries
+    here? They'd be exactly the same as the primary...unless the metadata or
+    maximum_size has changed since the dataset was migrated between nodes.
+    And without a change to the ``datasets`` schema, there's no way to mark a
+    dataset as secondary. Perhaps there needs to be a current_node key...or
+    something? We need to understand how this API is likely to be used.
```

Comment:
Probably something like `primary=..., replicas=[...]`.  The other option would be to re-structure this as an object with node addresses as property names and dataset lists as property values, making location implicit instead of an explicit part of the dataset object.  These are isomorphic though and probably each caters to some use cases.  The former keeps the data structures the same shape so I'm inclined to lean that way.

This is all out of scope for this issue, though, since it's still impossible to have replicas with Flocker.